exporter/sumologicexporter/sender.go

```diff
@@ -0,0 +1,262 @@
+
+package sumologicexporter
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"go.opentelemetry.io/collector/component/componenterror"
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+type sender struct {
+	buffer []pdata.LogRecord
+	config *Config
+	client *http.Client
+	filter *filtering
+}
+
+func newSender(cfg *Config, cl *http.Client, f *filtering) *sender {
+	return &sender{
+		config: cfg,
+		client: cl,
+		filter: f,
+	}
+}
+
+// Send sends data to sumologic
+func (s *sender) send(pipeline PipelineType, body io.Reader, fields FieldsType) error {
+	// Add headers
+	req, err := http.NewRequest(http.MethodPost, s.config.URL, body)
+	if err != nil {
+		return err
+	}
+
+	req.Header.Add("X-Sumo-Client", s.config.Client)
+
+	if len(s.config.SourceHost) > 0 {
+		req.Header.Add("X-Sumo-Host", s.config.SourceHost)
+	}
+
+	if len(s.config.SourceName) > 0 {
+		req.Header.Add("X-Sumo-Name", s.config.SourceName)
+	}
+
+	if len(s.config.SourceCategory) > 0 {
+		req.Header.Add("X-Sumo-Category", s.config.SourceCategory)
+	}
+
+	switch pipeline {
+	case LogsPipeline:
+		req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+		req.Header.Add("X-Sumo-Fields", string(fields))
+	case MetricsPipeline:
+		// ToDo: Implement metrics pipeline
+		return errors.New("Current sender version doesn't support metrics")
+	default:
+		return errors.New("Unexpected pipeline")
+	}
+
+	resp, err := s.client.Do(req)
+	// ToDo: Add retries mechanism
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode < 200 || resp.StatusCode >= 400 {
+		return fmt.Errorf("Error during sending data: %s", resp.Status)
+	}
+	return nil
+}
+
+func (s *sender) sendLogs(fields FieldsType) (int, error) {
+	switch s.config.LogFormat {
+	case TextFormat:
+		return s.sendLogsTextFormat(fields)
+	case JSONFormat:
+		return s.sendLogsJSONFormat(fields)
+	default:
+		return s.count(), errors.New("Unexpected log format")
+	}
+}
+
+func (s *sender) sendLogsTextFormat(fields FieldsType) (int, error) {
+	var (
+		body              strings.Builder
+		errs              []error
+		droppedTimeSeries int
+		currentTimeSeries int
+	)
+
+	for _, record := range s.buffer {
+		sent, appended, err := s.appendAndSend(record.Body().StringVal(), LogsPipeline, &body, fields)
+		if err != nil {
+			errs = append(errs, err)
+			if sent {
+				droppedTimeSeries += currentTimeSeries
+			}
+
+			if !appended {
+				droppedTimeSeries++
+			}
+		}
+
+		// If data was sent, cleanup the currentTimeSeries counter
+		if sent {
+			currentTimeSeries = 0
+		}
+
+		// If log has been appended to body, increment the currentTimeSeries
+		if appended {
+			currentTimeSeries++
+		}
+	}
+
+	err := s.send(LogsPipeline, strings.NewReader(body.String()), fields)
+	if err != nil {
+		errs = append(errs, err)
+		droppedTimeSeries += currentTimeSeries
+	}
+
+	if len(errs) > 0 {
+		return droppedTimeSeries, componenterror.CombineErrors(errs)
+	}
+	return droppedTimeSeries, nil
```

Comment:
So.... when there were no errors during logs sending we return `droppedTimeSeries`?
ditto in `sendLogsJSONFormat`

Reply:
it's going to be zero anyway. IMHO thats better approach because if sth goes wrong will be easier catchable by the tests

Reply:
A comon approach for other processors/exporters in OTC is just to update the relevant observability counter (essentially, a metric)